src/darwin/Framework/CHIP/MTRDeviceControllerFactory.mm

```diff
@@ -88,6 +90,20 @@ @interface MTRDeviceControllerFactory ()
 @property () chip::Credentials::DeviceAttestationVerifier * deviceAttestationVerifier;
 @property (readonly) BOOL advertiseOperational;
 @property (nonatomic, readonly) Credentials::IgnoreCertificateValidityPeriodPolicy * certificateValidityPolicy;
+// Lock used to serialize access to the "controllers" array, since it needs to
+// be touched from both whatever queue is starting controllers and from the
+// Matter queue.  The way this lock is used assumes that:
+//
+// 1) The only mutating accesses to the controllers array happen from outside
+//    the Matter queue (which is a good assumption, because those functions do
+//    sync dispatch to the Matter queue).
```

Comment:
nit: Can we make that comment clearer. I end up reading it multiple times to make sure I got it because of the sentence: "from outside there Matter queue (which is a good assumption, because those functions do sync dispatch to the Matter queue)".
I know what you mean but I found it confusing. Maybe just add "internally" ? "because those functions internally do sync dispatch to the Matter queue". It could also just be me, feel free to ignore if that the case.

Reply:
I'll see about rephrasing this.  I want this part to be very very clear, because the invariants here are kind of important.